akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -703,17 +746,38 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
       // to support manual join when joining to seed nodes is stuck (no seed nodes available)
       stopSeedNodeProcess()
 
-      if (address == selfAddress) {
-        becomeInitialized()
-        joining(selfUniqueAddress, cluster.selfRoles, cluster.settings.AppVersion)
-      } else {
-        val joinDeadline = RetryUnsuccessfulJoinAfter match {
-          case d: FiniteDuration => Some(Deadline.now + d)
-          case _                 => None
+      val appVersionOpt = laterAppVersion match {
+        case None => Some(cluster.settings.AppVersion)
+        case Some(promise) =>
+          promise.future.value match {
+            case Some(Success(version)) => Some(version)
+            case Some(Failure(exc)) =>
+              logError("Can't join because later appVersion was completed with failure: {}", exc)
+              None
+            case None =>
+              import akka.pattern.pipe
+              // easiest to just try again via JoinTo when the promise has been completed
```

Comment:
Debug logging or something so it is possible to notice waiting for it is what is going on?

Reply:
👍 added in https://github.com/akka/akka/pull/31934/commits/49b339ab6b91c8761a3c65f38ee44ca485c8ffab